snippets/PasswordRevealer.md

```diff
@@ -0,0 +1,45 @@
+### PasswordRevealer
+
+Renders a password input field with a reveal button.
+
+Initially set `state.shown` to `false` to ensure that the password is not shown by default.
+Create a method, `toggleShown`, which uses `Component.prototype.setState` to change the input's state from shown to hidden and vice versa, bind it to the component's context.
+In the`render()` method, use a`<div>` to wrap both the`<input>` and the `<button>` element that toggles the type of the input field
+Finally, bind the `<button>`'s `onClick` event to the `toggleShown` method.
+
+```jsx
+class PasswordRevealer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      shown: false
+    };
+    this.toggleShown = this.toggleShown.bind(this);
+  }
+
+  toggleShown() {
+    this.setState(state => ({ shown: !state.shown }));
+  }
+
+  render() {
+    return (
+      <div>
+        <input
```

Comment:
Should the be an unhandled input?

Reply:
We're not doing anything with it, but if you have any suggestions, please fire away!

Reply:
It's a password field, usually they are empty by default and then you can use the button to see what the heck you've typed. It would look weird to have some data in there by default.

Reply:
Well it is an unhandled input so it should populate with a value...

Reply:
What would the real-world use-case be for a password field prepopulated with some value? Isn't that a bad idea in general?

Reply:
No I mean when a user types in it it will accept letters, versus a strict value which will not. Unhandled inputs let that happen.

Reply:
@skatcat31 Can you give an example of what you mean? I am probably missing something here.

Reply:
unhandled element:

1. User types something in box, it shows up

handled element(assigned a value)

1. User types something and it goes to the ethos

handled element(assigned a value and a handler properly)

1. User types something and it shows up

Since this is an unhandled element, then it should work the way expected and input from user who can then reveal it will work.